grpc/src/main/java/io/stargate/grpc/service/Service.java

```diff
@@ -432,14 +435,24 @@ private void executePrepared(
                     Response.Builder responseBuilder = makeResponseBuilder(result);
                     switch (result.kind) {
                       case Void:
-                      case SchemaChange: // Fallthrough intended
+                        // fill tracing id for queries that doesn't return any data (i.e. INSERT)
+                        handleTraceId(
+                            result.getTracingId(),
+                            query.getParameters().getTracing(),
+                            responseBuilder);
+                        break;
+                      case SchemaChange:
                         break;
                       case Rows:
                         responseBuilder.setResultSet(
                             Payload.newBuilder()
                                 .setType(query.getValues().getType())
                                 .setData(
                                     handler.processResult((Rows) result, query.getParameters())));
+                        handleTraceId(
+                            result.getTracingId(),
+                            query.getParameters().getTracing(),
+                            responseBuilder);
```

Comment:
Could we use simple parameters here and move the logic of extracting sub-values to `handleTraceId`? This would reduce code duplication (cf. lines 439-442), WDYT?

Reply:
There are two types of parameters: `QueryParameters` AND `BatchParameters` so we cannot just pass parameters there. I can extract two helper methods that are extracting `tracingId`. 
Done in:
https://github.com/stargate/stargate/pull/1062/commits/231e839134e9ad213b725dba7bbaebc2eafe5d7e